### PR TITLE
coconutbattery: init at 3.9.12

### PR DESCRIPTION
**Pull request by stepbrobd**
###### Description of changes

added new darwin only package, coconutbattery https://www.coconut-flavour.com/coconutbattery

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by StepBroBD** [comment]
On aarch64-darwin and x86_64-darwin:

```shell
$ nix run nixpkgs#nixpkgs-review -- pr 240621
1 package added:
coconutbattery (init at 3.9.12)
1 package built:
coconutbattery
```

File: pkgs/os-specific/darwin/coconutbattery/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+, stdenvNoCC
+, fetchzip
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "coconutbattery";
+  version = "3.9.12";
+
+  src = fetchzip {
+    url = "https://coconut-flavour.com/downloads/coconutBattery_${lib.strings.stringAsChars (x: if x == "." then "" else x) version}.zip";
```
Comment (natsukium) [inline pkgs/os-specific/darwin/coconutbattery/default.nix:11]:
```suggestion
    url = "https://coconut-flavour.com/downloads/coconutBattery_${builtins.replaceStrings [ "." ] [ "" ] version}.zip";
```
I think `builtins.replaceStrings` is often used for such replacements.

**Comment by StepBroBD** [comment]
Thanks for the review! Could you also take a look at these?
- https://github.com/NixOS/nixpkgs/pull/240618
- https://github.com/NixOS/nixpkgs/pull/240625

File: pkgs/os-specific/darwin/coconutbattery/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, stdenvNoCC
+, fetchzip
+}:
+
+stdenvNoCC.mkDerivation rec {
```
Comment (natsukium) [inline pkgs/os-specific/darwin/coconutbattery/default.nix:6]:
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
```

File: pkgs/os-specific/darwin/coconutbattery/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  pname = "coconutbattery";
+  version = "3.9.12";
+
+  src = fetchzip {
+    url = "https://coconut-flavour.com/downloads/coconutBattery_${builtins.replaceStrings [ "." ] [ "" ] version}.zip";
+    sha256 = "sha256-8WxGjZbxUqchSIfvpK2RLifn7/TD5nau5hgMzLUiV2o=";
```
Comment (natsukium) [inline pkgs/os-specific/darwin/coconutbattery/default.nix:12]:
```suggestion
    hash = "sha256-8WxGjZbxUqchSIfvpK2RLifn7/TD5nau5hgMzLUiV2o=";
```

**Review by natsukium — APPROVED** [review]
I tried building and running it, and it seems to work fine.

As @drupol says [here](https://github.com/NixOS/nixpkgs/pull/240618#pullrequestreview-1506573061), the following changes could be a good idea.

File: pkgs/os-specific/darwin/coconutbattery/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+, stdenvNoCC
+, fetchzip
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "coconutbattery";
+  version = "3.9.12";
+
+  src = fetchzip {
+    url = "https://coconut-flavour.com/downloads/coconutBattery_${builtins.replaceStrings [ "." ] [ "" ] version}.zip";
```
Comment (drupol) [inline pkgs/os-specific/darwin/coconutbattery/default.nix:11]:
```suggestion
    url = "https://coconut-flavour.com/downloads/coconutBattery_${builtins.replaceStrings [ "." ] [ "" ] finalAttrs.version}.zip";
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — CHANGES_REQUESTED** [review]
LGTM now, could you please test it and don't forget to squash all your commits so you only have 1 single commit ?

**Review by drupol — APPROVED** [review]
LGTM now!

Let's wait for the end of the eval and I'll merge this.

Feel free to ping me when it's done.

**Pull request merged**